src/remediations/fifi.js

```diff
@@ -43,6 +50,187 @@ exports.checkSmartManagement = async function (remediation, smart_management) {
+        executor_id: exports.generateUuid(),
+        executor_name: 'Direct connected',
+        status: rhcRun.status,
+        system_count: rhcRun.system_count,
+        playbook_run_id: rhcRun[0].labels['playbook-run'],
+        playbook: rhcRun[0].url,
+        updated_at: rhcRun[0].updated_at,
+        count_failure: rhcRun.count_failure,
+        count_success: rhcRun.count_success,
+        count_running: rhcRun.count_running,
+        count_pending: 0, // RHC does not return the status pending
+        count_canceled: 0 // RHC does not currently return status canceled
+    });
+}
+
+function pushRHCRun (rhcRun, remediation, remediation_id, created_by) {
+    remediation.playbook_runs.push({
+        id: rhcRun[0].labels['playbook-run'],
+        status: rhcRun.status,
+        remediation_id,
+        created_by,
+        created_at: rhcRun[0].created_at,
+        updated_at: rhcRun[0].updated_at,
+        executors: []
+    });
+
+    const satRun = _.find(remediation.playbook_runs, satRun => satRun.id === rhcRun[0].labels['playbook-run']);
+    pushRHCExecutor(rhcRun, satRun);
+}
+
+exports.getRHCRuns = async function (playbook_run_id = null) {
+    const filter = createDispatcherRunsFilter(playbook_run_id);
+    const rhcRuns = await dispatcher.fetchPlaybookRuns(filter, RUNSFIELDS);
+
+    return rhcRuns;
+};
+
+exports.getRunHosts = async function (playbook_run_id) {
+    const runsFilter = createDispatcherRunsFilter(playbook_run_id);
+    const rhcRunHosts = await dispatcher.fetchPlaybookRuns(runsFilter, RUNSFIELDS);
+
+    return rhcRunHosts;
+};
+
+exports.getRunHostDetails = async function (playbook_run_d, system_id) {
+    const runsFilter = createDispatcherRunHostsFilter(playbook_run_d);
+    const runHostsFilter = createDispatcherRunHostsFilter(playbook_run_d, system_id);
+    const rhcRunHosts = await dispatcher.fetchPlaybookRuns(runsFilter, RUNSFIELDS);
+    const rhcRunHostDetails = await dispatcher.fetchPlaybookRunHosts(runHostsFilter, RUNHOSTFIELDS);
+
+    if (!rhcRunHosts || !rhcRunHostDetails) {
+        return null;
+    }
+
+    const host = _.find(rhcRunHosts.data, host => host.id === system_id);
+
+    return formatRHCHostDetails(host, rhcRunHostDetails);
+};
+
+exports.combineHosts = function (rhcRunHosts, systems) {
+    rhcRunHosts = exports.formatRunHosts(rhcRunHosts);
+
+    _.forEach(rhcRunHosts, host => {
+        pushRHCSystem(host, systems);
+    });
+};
+
+exports.combineRuns = function (rhcRuns, remediation, remediation_id, created_by) {
+    if (!rhcRuns) {
+        return remediation.playbook_runs;
+    }
+
+    rhcRuns = formatRHCRuns(rhcRuns);
+
+    if (!remediation) {
+        _.forEach(rhcRuns, rhcRun => {
+            pushRHCRun(rhcRun, remediation, remediation_id, created_by);
+        });
+    }
+
+    _.forEach(rhcRuns, rhcRun => {
+        const satRun = _.find(remediation.playbook_runs, run => run.id === rhcRun[0].labels['playbook-run']);
+        if (_.isUndefined(satRun)) {
```

Comment:
I did not know you're not storing a playbook_run entry in case of RHC-only runs. Have you considered storing it always?